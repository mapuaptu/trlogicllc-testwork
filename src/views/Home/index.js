import Button from '@/components/Button/index.vue';
import ModalCreateNode from '@/views/Modals/CreateNote/index.vue';
import { mapState } from 'vuex';

export default {
  name: 'HomeView',
  components: {
    Button,
  },
  computed: {
    ...mapState({ notes: 'notes' }),
  },
  methods: {
    addNote() {
      return this.$modal.show(
        ModalCreateNode,
        { title: 'Create note' },
        { height: 'auto' },
      );
    },
  },
};
