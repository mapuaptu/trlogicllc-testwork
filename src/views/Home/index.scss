@import '@/styles/_variables.scss';

.home {
  position: relative;
}
