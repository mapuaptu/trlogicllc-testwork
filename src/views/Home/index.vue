<template>
  <div :class="$style.home">
    <div :class="$style.controls">
      <Button
        icon="plus-outline"
        @click.native="addNote"
      >
        Add todo
      </Button>

      <Button
        icon="delete-outline"
        @click.native="deleteAllNotes"
      >
        Delete all
      </Button>
    </div>

    <div>
      <div
        v-for="(item, index) in notes"
        :key="index"
      >
        {{ item.name }}
      </div>
    </div>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
