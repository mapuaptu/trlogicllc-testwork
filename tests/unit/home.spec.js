import Home from '@/views/Home/index.vue';
import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

const localVue = createLocalVue();

localVue.use(Vuex);

const store = new Vuex.Store({
  state: {
    notes: [],
  },
});

describe('Home component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(Home, { store, localVue });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
