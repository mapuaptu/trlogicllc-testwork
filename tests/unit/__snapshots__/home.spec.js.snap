// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view 1. Rendered success 1`] = `
<div>
  <div>
    <div class=""><span>
      Add note
    </span>
      <div><svg width="24" height="24" viewbox="0 0 24 24">
          <path d="M4,9H9V4H15V9H20V15H15V20H9V15H4V9M11,13V18H13V13H18V11H13V6H11V11H6V13H11Z" fill="#fff"></path>
        </svg></div>
    </div>
  </div>
  <div></div>
</div>
`;
